### ISSUE TITLE
session-control not working when device is locked

### ISSUE THREAD

**Issue by ashokraghu14** [LaeraFelipe/react-session-control#1]
When the system is locked, the popup shows and says, you're idle and you'll be logged out, when it's not logging out until the user opens his system. @LaeraFelipe  Please help me out is there any way to sign-out the user even when his system is locked.

**Comment by LaeraFelipe** [comment]
Hi, to logout the user you should remove the token from storage.

**Comment by ashokraghu14** [comment]
Hi, it's working when the user is idle, but it's the problem when the user is idle and he locked his system, so it shows the pop-up, but only when he opens his system, the timeout progress bar start to decrease and eventually logs out.

**Comment by LaeraFelipe** [comment]
Could you post the parts of the code where you use the UnauthenticatedSessionControl and AuthenticatedSessionControl components and the handlers for onLogin and onLogout?

**Comment by ashokraghu14** [comment]
Here's my code:

```
this.state = {
     inactivityTimeoutCount: 1800
}

function handleSessionControlLogin() {
  if (LogoutTypes.inactivity == "inactivity") {
    localStorage.clear();
    window.location = "/login";
  }
}

<div className="wrapper-main">
          <AuthenticatedSessionControl
            inactivityTimeout={this.state.inactivityTimeoutCount}
            modalInactivityTimeout={60}
            storageTokenKey="userToken"
            onLogout={logout => handleSessionControlLogin(logout)}
          />
```

**Comment by LaeraFelipe** [comment]
The component AuthenticatedSessionControl must be placed only in an authenticated area. In an unauthenticated context you should use UnauthenticatedSessionControl. Your problem must be occurring because even though you are logged out you are rendering the AuthenticatedSessionControl.

There is also a problem in your handleSessionControlLogin function. When you compare (LogoutTypes.inactivity == "inactivity") you will always get true because you are comparing the value of a constant with its literal value. You must use the value from the parameter, like this:

```
function handleSessionControlLogin(logoutType) {
  if (logoutType === LogoutTypes.inactivity ) {
    localStorage.clear();
    window.location = "/login";
  }
}

```

**Comment by LaeraFelipe** [comment]
@ashokraghu14 I'm sorry, I believe I understand your problem now. The browser is probably not running while the device is locked. I will work on a fix to check the difference between the user's last action and the time the device was unlocked.

**Comment by ashokraghu14** [comment]
@LaeraFelipe Thank you. Please update the bug, when you find the solution.

**Comment by LaeraFelipe** [comment]
@ashokraghu14 A fix for this problem was published in version 2.0.11.

**Comment by ashokraghu14** [comment]
Hi @LaeraFelipe, thanks for the fix. It's now working when the device is locked, but it's not working when the device is on sleep. Please check is there any fix for that too. Thanks in advance.
